### PR TITLE
Bam Week 5: Cluedo card generator 

### PR DESCRIPTION
**Pull request by BamStugendal**
The first steps were relatively easy but for a more advanced level, I had a harder time wrapping my head around. Like how the logic and what I needed to do to make the functions. If I had more time I would like to write the code so it isn’t so buggy. Because right now you have to click the card in the right order for it to work and so on. 

**Live link**: https://cluedo-generator.netlify.app/code/

### CODE REVIEW

File: code/clue.js [hunk 1]
```diff
@@ -7,53 +6,356 @@ const mrGreen = {
...
+  place: null
+};
+
+const pistol = {
+  name: 'Pistol',
+  weight: 9,
+  id: 'pistol',
+  favoritePlace: 'Study',
+  place: null
+};
```
Comment (JHarde) [inline code/clue.js:147]: Good and clear display of the objects, well done! 👍

File: code/clue.js [hunk 2]
```diff
@@ -7,53 +6,356 @@ const mrGreen = {
...
+  roomInfo.style.display = 'block';
+
+  shuffleRoom(weapons);
+  mystery.room = mystery.weapon.place;
+
+  roomName.innerHTML = `${mystery.room}`
+};
+
+
+const revealValdidation = () => {
```
Comment (JHarde) [inline code/clue.js:317]: Nice feature adding the validation so you have to pick all three cards to move on to the next page! 😃

File: code/clue.js [hunk 3]
```diff
@@ -7,53 +6,356 @@ const mrGreen = {
...
+}; 
+
+
+const revealMystery = () => {
+  mysteryText.innerHTML = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} with a ${mystery.weapon.name}.`;
+};
+
+
+//reset
+const resetCards = () => {
```
Comment (JHarde) [inline code/clue.js:332]: Well done with the reset button! 👍

File: code/clue.js [hunk 4]
```diff
@@ -7,53 +6,356 @@ const mrGreen = {
...
 
-// This function will be invoked when you click on the killer card.
+const populateArray = (elementToPopulate, howManyTimes) => {
+  for (let i=0; i<howManyTimes; i++) {
+    return elementToPopulate
+  };
+};
+
+
+const shuffleWeapon = (arrayOfSuspects) => {
```
Comment (JHarde) [inline code/clue.js:238]: Nice job with the shuffling functions! Was there supposed to be a button in which the user could manually change the favourite weapon? 🤔

File: code/style.css [hunk 5]
```diff
@@ -1,58 +1,147 @@
...
+}
+
+.lds-ring {
+  display: none;
+  position: relative;
+  margin-top: 27%;
+  width: 80px;
+  height: 80px;
+}
+.lds-ring div {
```
Comment (JHarde) [inline code/style.css:121]: Supercool loading function! 🥇

File: code/style.css [hunk 6]
```diff
@@ -1,58 +1,147 @@
...
+  cursor: pointer;
+  margin-bottom: 20px;
+}
+
+button:hover {
+  background-color: #A6C8A6;
+  color: #fff;
+  border: rgb(27, 27, 27) 3px solid;
+}
+
```
Comment (JHarde) [inline code/style.css:107]: Cool hover feature!

File: code/style.css [hunk 7]
```diff
@@ -1,58 +1,147 @@
...
   height: 400px;
-  background: midnightblue;
+  background: #EAC76B;
 
   text-align: center;
   padding-top: 30px;
   border-radius: 16px;
+  margin: 15px;
+  cursor: pointer;
 }
```
Comment (JHarde) [inline code/style.css:85]: Nice with different colors on the decks, makes it easy to spot the difference between them :)

File: code/index.html [hunk 8]
```diff
@@ -4,34 +4,71 @@
...
+    <link href="https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,400;0,500;0,700;0,900;1,400;1,700&display=swap" rel="stylesheet">
     <link rel="stylesheet" href="style.css" />
-    <title>Technigo Project Cluedo</title>
+    <link rel="icon" type="image/png" href="assets/glas.png">
+    <title>Cluedo</title>
   </head>
   <body>
+    <section class="start-text">
+      <h1>One murder... 6 suspects</h1>
+    </section>
```
Comment (JHarde) [inline code/index.html:15]: I like that you wrapped the h1 in a section!

File: code/index.html [hunk 9]
```diff
@@ -4,34 +4,71 @@
...
+          <p id="favoriteWeapon"></p>
+          <p id="killerAge"></p>
+          <img id="killerImage" src="" height="200" />
+          <p id="killerDescription"></p>
+         </div>
+        <div class="lds-ring" id="killerLoading">
+          <div></div>
+          <div></div>
+          <div></div>
+          <div></div>
```
Comment (JHarde) [inline code/index.html:31]: What are the empty divs used for? 🤔

File: code/index.html [hunk 10]
```diff
@@ -4,34 +4,71 @@
...
+    </section>
     <section class="cards">
       <article class="killer-deck" id="killerCard">
-        <p>KILLER</p>
-        <p id="killerName"></p>
-        <p id="killerAge"></p>
-        <p id="killerOccupation"></p>
-        <img id="killerImage" src="" height="100" />
-        <p id="killerDescription"></p>
+        <h2>KILLER</h2>
```
Comment (JHarde) [inline code/index.html:18]: Like how you changed "killer" from <p> tag to <h2>

**Review by JHarde — APPROVED** [review]

**Review by JHarde — APPROVED** [review]
Really nice design of the page, makes me want to play Cluedo more often! Like how clear the code is, makes total sense! Can see that you aimed high and you definitely did a great job! Only thing I'm missing is a button for changing the favourite weapon so that it gets displayed for the user. Yet again, great job, be proud of yourself! 🏅

File: code/clue.js [hunk 11]
```diff
@@ -7,53 +6,356 @@ const mrGreen = {
...
+  'Living Room', 
+  'Observatory', 
+  'Theater', 
+  'Guest House', 
+  'Patio'
+];
+
+let mystery = {
+  killer: null,
+  weapon: null,
```
Comment (EmOlsz) [inline code/clue.js:190]: Very nice initialisation of object, it's the most semantic way to use `null` values :)

Moreover, you can use `const` keyword over here - we will be changing values of properties, that's true, but JavaScript will not consider this as overwriting. What counts for JavaScript is that object will stay object

File: code/clue.js [hunk 12]
```diff
@@ -7,53 +6,356 @@ const mrGreen = {
...
+const revealButton = document.getElementById('reveal');
+const resetButton = document.getElementById('reset');
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
+
+// returns the provided suspects favourite weapon as a weapon obj
+const favoriteWeapon = (suspect) => {
+  return weapons.find((weapon) => {return weapon.id === suspect.favouriteWeapon});
+}
```
Comment (EmOlsz) [inline code/clue.js]: We can simplify that :

```
const favoriteWeapon = suspect => weapons.find(weapon => weapon.id === suspect.favouriteWeapon);
```

File: code/clue.js [hunk 13]
```diff
@@ -7,53 +6,356 @@ const mrGreen = {
...
+
+  shuffleRoom(weapons);
+  mystery.room = mystery.weapon.place;
+
+  roomName.innerHTML = `${mystery.room}`
+};
+
+
+const revealValdidation = () => {
+  if (mystery.killer === null || mystery.weapon === null || mystery.room === null) {
```
Comment (EmOlsz) [inline code/clue.js]: We can simplify that :

```
if (!mystery.killer || ... ) { ... }
```

For further reference check out https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/

File: code/index.html [hunk 14]
```diff
@@ -4,34 +4,71 @@
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <link href="https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,400;0,500;0,700;0,900;1,400;1,700&display=swap" rel="stylesheet">
     <link rel="stylesheet" href="style.css" />
```
Comment (EmOlsz) [inline code/index.html]: Let's not forget about `./` in path to file

File: code/style.css [hunk 15]
```diff
@@ -1,58 +1,147 @@
...
+  background-color: rgb(27, 27, 27);
+}
+
+.start-text {
+  font-weight: 900;
+  font-size: 100px;
+  text-align: center;
+  padding: 30px;
+  margin-top: 30px;
+  margin-bottom:  -100px;
```
Comment (EmOlsz) [inline code/style.css]: It's not the best idea to use margin with values lower than zero. Aren't there better options?

File: code/style.css [hunk 16]
```diff
@@ -1,58 +1,147 @@
...
   margin: 50px auto;
   width: 90%;
   height: 600px;
-  background: #ccc;
   display: flex;
   justify-content: space-around;
   align-items: center;
+  border-radius: 30px;
+  max-width: 1200px;
+
```
Comment (EmOlsz) [inline code/style.css]: Unnecessary blank line

File: code/style.css [hunk 17]
```diff
@@ -1,58 +1,147 @@
...
+  width: 64px;
+  height: 64px;
+  margin: 8px;
+  border: 8px solid #fff;
+  border-radius: 50%;
+  animation: lds-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
+  border-color: #fff transparent transparent transparent;
+}
+.lds-ring div:nth-child(1) {
+  animation-delay: -0.45s;
```
Comment (EmOlsz) [inline code/style.css:134]: Good job :)

**Review by EmOlsz — CHANGES_REQUESTED** [review]
Congratulations, great job! A lot of extra requirements done :) I'm happy to see such advanced code.
Still, there is couple of things you could fix / improve - please, take a look on my comments :) After that, I'll be able to approve your project

**Review by EmOlsz — APPROVED** [review]

**Pull request closed without merging**